modules/storages/app/common/storages/peripherals/storage_interaction/request_url_builder.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal:true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module Peripherals
+    module StorageInteraction
+      class RequestUrlBuilder
+        class << self
+          def build(storage, *path_fragments)
+            host_uri = storage.uri
+
+            URI.join(host_uri.origin,
+                     ensure_sub_path(host_uri.path),
+                     *split_and_escape(path_fragments))
+               .to_s
+          end
+
+          def path(*path_fragments)
```

Comment:
I think the name is not great. It is a noun, so seems like a property of object, but it is not called on an object. I suggest to have the name to be an action describing thing like: `escape_and_join`, `join`, `join_paths`. Something of that kind. What do you think?

Reply:
I disagree with a noun being a bad method name. We have a ton of methods using nouns, because they do return a specific entity, object or constructed representation of a domain thing. And the noun is the description of that. Common examples:

- location
- uri
- header
- success
- failure

I'd like to keep the name, better yet, I'd like to rename `build` to `url` to make it consistent and more distinguishable.

`RequestUrlBuilder.url()`
`RequestUrlBuilder.path()`

Reply:
It is still feels somehow uneasy to me, but there is no problem. Let's have it like that :+1: